code/modules/mob/interactive.dm

```diff
@@ -865,7 +867,7 @@
 	if(prob(10)) // 10% chance to broadcast it over the radio
 		chatmsg = ";"
 
-	if(prob(chattyness) || knownStrings.len < 10) // say a generic phrase, otherwise draw from our strings.
+	if(prob(chattyness) && knownStrings.len < 10) // say a generic phrase, otherwise draw from our strings.
```

Comment:
You're going about it the wrong way @bgobandit.

Instead of setting the chattyness to 0, remove the chat module from the functions list, which will completely disable all chat functions.